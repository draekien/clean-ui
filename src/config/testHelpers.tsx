import React, { ReactNode } from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '../components/theme/theme.provider';

export const themedRender = (element: ReactNode) =>
  render(<ThemeProvider>{element}</ThemeProvider>);
