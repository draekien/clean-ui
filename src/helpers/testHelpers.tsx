import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '../components/theme/theme.provider';

export const themedRender = (element: React.ReactNode) => {
  return render(<ThemeProvider>{element}</ThemeProvider>);
};

export const hexToRgb = (hex: string) => {
  hex = hex.replace(/[^0-9A-F]/gi, '');
  const bigint = parseInt(hex, 16);
  const r = (bigint >> 16) & 255;
  const g = (bigint >> 8) & 255;
  const b = bigint & 255;

  return `rgb(${r}, ${g}, ${b})`;
};
