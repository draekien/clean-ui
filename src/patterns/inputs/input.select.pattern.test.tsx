import React from 'react';
import { themedRender } from '../../helpers/testHelpers';
import { SelectInput } from './input.select.pattern';

const rows = [
  'Travis Willingham',
  'Laura Bailey',
  'Ashleigh Johnson',
  'Sam Riegal',
  'Matthew Mercer',
  'Marisha Ray',
  "Liam O'Brian",
  'Talisen Jaffe',
  'Percival Fredrickstein von Musel Klossowski de Rolo III',
];

describe('The select input pattern', () => {
  it('Should match snapshot when mode is single', () => {
    const { container } = themedRender(
      <SelectInput
        inputId="test"
        value="Travis Willingham"
        items={{ rows }}
        label="test"
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot when mode is multiple', () => {
    const { container } = themedRender(
      <SelectInput
        inputId="test"
        value="Travis Willingham"
        items={{ rows }}
        label="test"
        multiple
      />
    );

    expect(container).toMatchSnapshot();
  });
});
