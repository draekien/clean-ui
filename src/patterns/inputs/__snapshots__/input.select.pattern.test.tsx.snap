// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The select input pattern Should match snapshot when mode is multiple 1`] = `
<div>
  <div
    class="css-109th3q-InputWrapper"
  >
    <div
      class="css-1qm0f6g-InputLabel"
    >
      <label
        for="test"
      >
        <div
          class="css-14i3diu-Text css-vurnku"
        >
          test
        </div>
      </label>
    </div>
    <div
      class="css-apg3wl-SelectInput"
    >
      <div>
        <div
          class="css-1qnoh1k-DropdownButton"
        >
          <button
            class="css-3iue9s-Button"
            icon="expand_more"
          >
            <div
              class="css-1xz4u2o-Text css-vurnku"
            >
              1 item
            </div>
            <span
              class="css-lh6qdg-ButtonIcon"
            >
              <i
                class="material-icons  css-y0j37o-Icon"
              >
                expand_more
              </i>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="css-1539o05-SelectInput"
  >
    <span
      class="css-x2q8nv-Tag"
    >
      <span
        class="css-15nj5vr-Tag"
      >
        <div
          class="css-1su31x7-Text css-vurnku"
        >
          Travis Willingham
        </div>
      </span>
      <button
        class="css-7iz1i8-Tag"
      >
        <i
          class="material-icons  css-y0j37o-Icon"
        >
          close
        </i>
      </button>
    </span>
  </div>
</div>
`;

exports[`The select input pattern Should match snapshot when mode is single 1`] = `
<div>
  <div
    class="css-109th3q-InputWrapper"
  >
    <div
      class="css-1qm0f6g-InputLabel"
    >
      <label
        for="test"
      >
        <div
          class="css-14i3diu-Text css-vurnku"
        >
          test
        </div>
      </label>
    </div>
    <div
      class="css-apg3wl-SelectInput"
    >
      <div>
        <div
          class="css-1qnoh1k-DropdownButton"
        >
          <button
            class="css-3iue9s-Button"
            icon="expand_more"
          >
            <div
              class="css-1xz4u2o-Text css-vurnku"
            >
              Travis Willingham
            </div>
            <span
              class="css-lh6qdg-ButtonIcon"
            >
              <i
                class="material-icons  css-y0j37o-Icon"
              >
                expand_more
              </i>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
